samples/python/thermo/coverage_dependent_surf.py

```diff
@@ -0,0 +1,126 @@
+"""
+This example demonstrates (1). the four different dependency models available
+for coverage-dependent enthalpy and entropy calculations and (2). capability
+of including self-interaction but also cross-interaction among different
+surface species.
+
+First demonstration is with a binary system composed of Pt and CO* where Pt
+is vacant Pt sites and CO* is CO adsorbates with CO*-CO* lateral interaction.
+Coverage-dependent enthalpy and entropy calculated with the four dependency
+models are plotted against discrete data points used for dependency model
+parametrization.
+
+Second demonstration is with a ternary system composed of Pt, CO* and O*.
+In this system, enthalpy of CO* is dependent both of CO* coverage and O*
+coverage. That is, it encompasses CO*-CO* interaction as well as CO*-O*
+interaction. The CO* enthalpy is plotted as a function of CO* and O*
+coverages.
+
+Requires: cantera >= 2.6.0, matplotlib >= 2.0, numpy >= 1.12
+Keywords: thermodynamics, surface, non-ideal adsorbate
+"""
+
+import cantera as ct
+import numpy as np
+import matplotlib.pyplot as plt
+
+# first demonstration: the four coverage dependency models
+
+# provide discrete enthalpy and entropy values calculated with DFT
+# array of CO* coverage
+dft_covs = [0., 0.11, 0.22, 0.33, 0.44, 0.56, 0.67, 0.78, 0.89, 1.0]
+# array of nondimensionalized DFT-derived CO* enthalpy values
+dft_hrts = [-112.8 , -112.8 , -108.78, -105.76, -106.11, -102.51,  -96.83,
+            -85.03,  -78.76,  -72.19]
+# array of nondimensionalized DFT-derived CO* entropy values
+dft_srs = [2.46, 2.46, 3.79, 3.28, 3.14, 2.41, 2.19, 1.63, 1.6 , 1.12]
+
+# create a dictionary to store coverage-dependent enthalpy and entropy
+# from the four dependency models. key is the model name and value is
+# a dictionary of yaml's phase name, enthalpy, and entropy
+models_dict = {'linear': {'phase_name': 'covdep_lin'},
+               'piecewise-linear': {'phase_name': 'covdep_pwlin'},
+               'polynomial': {'phase_name': 'covdep_poly'},
+               'interpolative': {'phase_name': 'covdep_int'}}
+
+# create an array of CO* coverages from 0 (no CO*) to 1 (full mono-layer)
+CO_covs = np.linspace(0, 1.0, 101)
+# remember CO* species index
+i_CO = 1
+
+# get coverage-dependent enthalpy and entropy for all four models
+for model in models_dict.keys():
```

Comment:
Iterating over the keys is default, so you can just write:
```suggestion
for model in models_dict:
```